packages/compass-query-bar/src/stores/query-changed-store.js

```diff
@@ -81,8 +83,25 @@ const configureStore = (options = {}) => {
         } else {
           debug('Error: AppRegistry not available for query-changed-store');
         }
+
         if (globalRegistry) {
-          globalRegistry.emit('compass:query-bar:query-changed', newState);
+          const dataService = this.queryBarStore.dataService;
+          const nsobj = toNS(newState.ns);
+          let collectionType = '';
+
+          if (dataService) {
+            const listCollections = util.promisify(dataService.listCollections.bind(dataService));
```

Comment:
unless the namespace changes the collection is going to be the same, and so is the type, do we call listCollections all the time a query changes?